lib/rules/no-unused-vars.js

```diff
@@ -153,9 +150,32 @@ module.exports = {
         }
 
         /**
+         * Determine if an identifier is used either in for...of or for...in loops.
+         * @param {Reference} ref - The reference to check.
+         * @returns {Boolean} whether reference is used in the for...of or for...in loops
```

Comment:
This is the convention we've used throughout to distinguish between boolean primitive values and boolean object values. It's the difference between:

```js
// boolean
var found = true;

// Boolean
var found = new Boolean(true);
```